pandas/tests/frame/test_sorting.py

```diff
@@ -116,9 +116,26 @@ def test_sort_values(self):
         self.assertRaises(ValueError, lambda: frame.sort_values(
             by=['A', 'B'], axis=2, inplace=True))
 
-        msg = 'When sorting by column, axis must be 0'
-        with assertRaisesRegexp(ValueError, msg):
-            frame.sort_values(by='A', axis=1)
+        # by row (axis=1): GH 10806
+        sorted_df = frame.sort_values(by=3, axis=1)
```

Comment:
this is what ``_get_axis_number(..)`` does

the doc-string should be something like:

```
axis : {index (0), columns (1)}
```

which we programatically generate (this is from ``DataFrame.min`` for example), look in generic.py to see how